torch/optim/adam.py

```diff
@@ -52,7 +52,7 @@ def __init__(self,
             # Support AMP with FP16/BF16 model params which would need
             # higher prec copy of params to do update math in higher prec to
             # alleviate the loss of information.
-            fused_supported_devices = _get_fused_kernels_supported_devices()
+            fused_supported_devices = _get_fused_kernels_supported_devices(self.__class__)
```

Comment:
hm...I have to think about whether this is the right abstraction, or if we want to do something like this instead
```suggestion
            fused_supported_devices = _get_fused_kernels_supported_devices() + ["cpu"]
```

Reply:
Updated with your advice, thanks.